scripts/sample_qc.py

```diff
@@ -0,0 +1,332 @@
+from gnomad.utils.annotations import bi_allelic_expr
+from gnomad.sample_qc.filtering import (
+    compute_stratified_sample_qc,
+    merge_sample_qc_expr,
+)
+from gnomad.sample_qc.sex import get_ploidy_cutoffs
+from gnomad.sample_qc.relatedness import (
+    compute_related_samples_to_drop,
+    DUPLICATE_OR_TWINS,
+    get_relationship_expr,
+    PARENT_CHILD,
+    SIBLINGS,
+    UNRELATED,
+)
+from gnomad.sample_qc.ancestry import pc_project
+
+from ccdg_qc.resources import *
+
+
+logging.basicConfig(format="%(levelname)s (%(name)s %(lineno)s): %(message)s")
+logger = logging.getLogger("CCDG_sample_qc")
+logger.setLevel(logging.INFO)
+
+
+def get_qc_vds(
+    data_type: str = "genomes",
+    pre_filter_variants: bool = False,
+    split: bool = False,
+    autosome_only: bool = False,
+    interval_qc: bool = False,
+) -> hl.vds.VariantDataset:
+    """
+    Wrapper function to get ccdg vds with desired filtering
+    :param data_type: Whether data is from genomes or exomes, default is genomes
+    :param split: Perform split on VDS, default is False
+    :param autosome_only: Whether to filter to variants in autosome, default is False
+    :param interval_qc: Whether to filter to high quality intervals for exomes QC, default is False
+    :return: ccdg vds with chosen filters
+    """
+    vds = hl.vds.read_vds(get_vds_path(data_type))
+    if (data_type == "exomes") & interval_qc:
+        logger.info("Filtering CCDG exomes VDS to high quality intervals...")
+        int_ht = hl.read_table(
+            get_ccdg_results_path(data_type=data_type, result="high_qual_intervals")
+        )
+        vds = hl.vds.filter_intervals(
+            vds, intervals=int_ht.interval.collect(), keep=True
+        )
+
+    if split:
+        logger.info("Splitting multi-allelic sites in CCDG %s VDS...", data_type)
+        vds = hl.vds.split_multi(vds, filter_changed_loci=True)
+
+    if autosome_only:
+        logger.info("Filtering CCDG %s VDS to autosomes...", data_type)
+        var_ht = filter_to_autosomes(vds.variant_data).rows()
+        vds = hl.vds.filter_variants(vds, var_ht, keep=True)
+
+    return vds
+
+
+def compute_sample_qc(data_type: str = "genomes") -> hl.Table:
+    """
+    Perform sample QC on the split VDS table using `compute_stratified_sample_qc`.
+    :param data_type: Whether data is from genomes or exomes, default is genomes
+    :return: Table containing sample QC metrics
+    :rtype: hl.Table
+    """
+    logger.info("Computing sample QC on CCDG %s VDS", data_type)
+    vds = get_qc_vds(
+        data_type=data_type,
+        autosome_only=True,
+        split=True,
+    )
+    sample_qc_ht = compute_stratified_sample_qc(
+        vds,
+        strata={
+            "bi_allelic": bi_allelic_expr(vds.variant_data),
+            "multi_allelic": ~bi_allelic_expr(vds.variant_data),
+        },
+        tmp_ht_prefix=get_ccdg_results_path(data_type=data_type, result="sample_qc")[
+            :-3
+        ],
```

Comment:
Probably collapse into one line. Took me a second to figure out what was happening here